[Term Entry] Python sets .issubset() #4613 (#4634)
Codecademy/docs: content/python/concepts/sets/terms/issubset/issubset.md

---
Title: '.issubset()'
Description: 'Returns True if all elements in a set exist in another; otherwise, it returns False.'
Subjects:
  - 'Code Foundations'
  - 'Computer Science'
Tags:
  - 'Collections'
  - 'Methods'
  - 'Sets'
CatalogContent:
  - 'learn-python-3'
  - 'paths/computer-science'
---

> The **`.issubset()`** method returns `True` if all elements in a set exist within another specified set. If any elements are not contained within the specified set, it returns `False`.

Reviewer: ```suggestion
The **`.issubset()`** method checks whether all elements in one set exist within another specified set. It returns `True` if this condition is met; otherwise, it returns `False`.
```

## Syntax

```pseudo

> set.issubset(set1)

Reviewer: ```suggestion
set1.issubset(set2)
```

```

> - `set`: The set whose elements are being checked for existence within `set1`.
> - `set1`: The set being searched for the presence of elements from the `set`.

Reviewer: ```suggestion
- `set1`: The set whose elements are being checked for existence within `set2`.
- `set2`: The set being searched for the presence of elements from the `set1`.
```

## Example

> 

Reviewer: ```suggestion

The following example demonstrate the usage of `.issubset()` method:

```

---
Title: ".issubset()"
Description: "Returns True if all elements in a set exist in another; otherwise, it returns False."
Subjects:
  - "Code Foundations"
  - "Computer Science"
Tags:
  - "Collections"
  - "Functions"
  - "Methods"
  - "Sets"
CatalogContent:
  - "learn-python-3"
  - "paths/computer-science"
---

The **`issubset()`** method returns True if all of the elements in a set exist within another given set. If all of the elements are not contained within the given set, it will return False instead.

## Syntax

```pseudo
set_A.issubset(set_B)

# Alternatively, a shorter syntax version can be used to do the same thing.
set_A <= set_B
```

## Example

```py
set_A = {"red", "blue", "green"}
set_B = {"pink", "yellow", "red", "blue", "green"}

print(set_A.issubset(set_B))
```

> This would print `True`, because `set_B` contains all of the elements in `set_B`:

Reviewer 2: ```suggestion
This would print `True`, because `set_B` contains all of the elements in `set_B`. It produces the following output:
```

> ```py

Reviewer 2: ```suggestion
```shell
```

True
```

## Codebyte Example

```codebyte/python
x = {"cat", "dog", "fish"}
y = {"eagle", "dog", "mouse", "fish", "gorilla", "cat"}
z = {"apple", "banana", "cherry"}

> print(x.issubset(y)) # True, because y contains all of the elements in x.
> print(x.issubset(z)) # False, because z doesn't contain all of the elements in x.

Reviewer 2: ```suggestion
print(x.issubset(y)) 
print(x.issubset(z)) 
```
